pkg/server/server.go

```diff
@@ -446,6 +477,18 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 	if err != nil {
 		return err
 	}
+	if s.proxyStrategy == "destHost" {
+		// if the proxyStrategy is "destHost", use the agentHost as the ID
+		agentHost, err := agentHost(stream)
+		if err != nil {
+			return err
+		}
+		if agentHost != "" {
+			// if the agentHost is set, assign it to the agentID
+			agentID = agentHost
```

Comment:
I'm not a fan of changing the agentID, especially if one was provided already. This seems to be only used for speeding up the backend selection, and it should be possible to optimize around there rather than the renaming.

@cheftako do you have any thoughts?

Reply:
+1, If need to make requests of the KAS about the agent, this is the identifier for those requests. Having the requests work initially (prior to this line) and then stop working (once this is executed is going to be very confusing. I realize we have no queries like that today but other similar backend managers are likely to need to make sure queries. (Eg. to lookup the related failure zone)